tracer/test/Datadog.Trace.Tests/SpanStatisticalTests.cs

```diff
@@ -4,141 +4,132 @@
+
+        private const int NumberOfBuckets = 20;
+        private const ulong NumberOfIdsToGenerate = 1_500_000;
 
         // Helper numbers for logging and calculating
-        private static decimal _bucketSizePercentage = 100 / _numberOfBuckets;
-        private static ulong _bucketSize = _maxId / (ulong)_numberOfBuckets;
+        private const double BucketSizePercentage = 100.0 / NumberOfBuckets;
+        private const ulong BucketSize = MaxId / NumberOfBuckets;
+
+        private static readonly Dictionary<ulong, int> GeneratedIds = new();
 
         private readonly ITestOutputHelper _output;
 
         public SpanStatisticalTests(ITestOutputHelper output)
         {
-            void BlastOff()
-            {
-                if (_generatedIds.Keys.Count > 0)
-                {
-                    return;
-                }
+            _output = output;
 
-                _output.WriteLine($"Starting key generation.");
-                var stopwatch = Stopwatch.StartNew();
+            if (GeneratedIds.Keys.Count > 0)
+            {
+                return;
+            }
 
-                // populate the dictionary for all tests
-                for (ulong i = 0; i < _numberOfIdsToGenerate; i++)
-                {
-                    var id = GenerateId();
+            _output.WriteLine("Starting key generation.");
+            var stopwatch = Stopwatch.StartNew();
 
-                    _generatedIds.TryGetValue(id, out var hitCount);
+            // populate the dictionary for all tests
+            for (ulong i = 0; i < NumberOfIdsToGenerate; i++)
+            {
+                var id = IdGenerator.NextUInt64();
 
-                    hitCount++;
+                GeneratedIds.TryGetValue(id, out var hitCount);
 
-                    _generatedIds[id] = hitCount;
-                }
+                hitCount++;
 
-                stopwatch.Stop();
-                _output.WriteLine($"It took {stopwatch.ElapsedMilliseconds / 1000d} seconds to generate {_numberOfIdsToGenerate} keys.");
+                GeneratedIds[id] = hitCount;
             }
 
-            _output = output;
-            BlastOff();
+            stopwatch.Stop();
+            _output.WriteLine($"It took {stopwatch.ElapsedMilliseconds / 1000d} seconds to generate {NumberOfIdsToGenerate} keys.");
         }
 
         [Fact]
         public void GeneratedIds_Contain_High_Numbers()
         {
-            var rangeBound = _maxId - _bucketSize;
-            var keysWithinRange = _generatedIds.Keys.Where(i => i >= rangeBound).ToList();
-            _output.WriteLine($"Found {keysWithinRange.Count()} above {rangeBound}, the top {_bucketSizePercentage}% of values.");
-            Assert.True(keysWithinRange.Count() > 0);
+            const ulong rangeBound = MaxId - BucketSize;
+            var keysWithinRange = GeneratedIds.Keys.Where(i => i >= rangeBound).ToList();
+            keysWithinRange.Should().NotBeEmpty();
```

Comment:
Alternative simpler version
```suggestion
            GeneratedIds.Keys.Should.Contain(i => i >= rangeBound);
```

Reply:
Thanks